streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java

```diff
@@ -268,7 +272,12 @@ public MetadataResponse fetchMetadata() {
             new MetadataRequest.Builder(null),
             Time.SYSTEM.milliseconds(),
             true);
-        final ClientResponse clientResponse = sendRequest(clientRequest);
+        ClientResponse clientResponse;
+        try {
+            clientResponse = sendRequest(clientRequest);
+        } catch (Exception e) {
+            throw new StreamsException("Failed to send request");
```

Comment:
Apart from a `StreamsException` i think the only other exception that `poll` is going to throw is `IllegalStateException` - should we just handle this in `sendRequest` and leave this as it was.
Even if there are more exceptions, i think it would be better to handle it in `sendRequest` and throw a `StreamsException` from there